addonscript/src/addonscript/api.py

```diff
@@ -14,6 +14,34 @@
 UPLOAD_STATUS_PK = "api/v3/addons/{id}/versions/{version}/uploads/{upload_pk}/"
 
 
+# https://addons-server.readthedocs.io/en/latest/topics/api/applications.html
+# XXX when we want to support multiple products, we'll need to unhardcode the
+#     `firefox` below
+ADD_VERSION = "api/v4/applications/firefox/{version}/"
+
+
+async def add_version(context, version):
+    """Add a new version to AMO.
+
+    Use the `min_version` here, rather than the string with the buildid etc.
+
+    Raises:
+        BadVersionError: If the XPI's version fails to sanity check against AMO
+
+
+    """
+    url = get_api_url(context, ADD_VERSION, version=version)
+
+    try:
+        await amo_put(context, url, data=None)
```

Comment:
@escapewindow I wasn't sure whether this is fine or not. I would've loved to log better in case of 201/202 to better debug later on if we have to. But `amo_put` strips the status code and returns the json directly. I can either duplicate that function and process the status code or keep as is for now.